components/list/list-item-checkbox-mixin.js

```diff
@@ -123,6 +125,13 @@ export const ListItemCheckboxMixin = superclass => class extends SkeletonMixin(s
 			}
 		}
 	}
+	_onMouseEnterSelection() {
+		this._hoveringSelection = !this.selectionDisabled;
```

Comment:
Since we are checking the `selectionDisabled` property here instead of in the CSS, if selection is disabled when the mouse enters the selection area, the `_hoveringSelection` property will not be set. Then, if selection is enabled, the selection would be hovered but the correct styles wouldn't be used. Or would the `mouse-enter` event be dispatched again when the component re-renders? Is that a case we need to cover?

Reply:
I don't think turning off pointer events fixes the problem. It introduces a similar issue, where we if `_hovering-selection` is set and then the selection is disabled, the `_hovering-selection` styles will never be removed.

It might be better to keep these `selection-disabled` checks in the CSS styles.

Reply:
I looked into that and saw that if hovering over the element when selection is disabled, the styles will only update after moving the mouse. This can become instant with a call to `willUpdate` to update `_hovering-selection` after changing `selection-disabled`, but that maybe too expensive just to handle that edge case.

I also saw that the same problem with `_hovering-primary-action`(even before these PR's changes) and I could apply the same changes there.

The main issue with setting the checks in the style is that they would need to be overridden in `d2l-list-item-button-mixin` and `d2l-list-item-link-mixin` which means a lot of repeated code and complex selections that initially caused the defects this PR tries to solve.

Reply:
I think using `willUpdate` to set `_hoveringSelection` to false sounds reasonable here then with your initial solution. It shouldn't be expensive as it would occur before the DOM updates so it should still only update once.